Lab04

noahstier opened the pull request.

noahstier committed:
initial lab04 formatting

noahstier committed:
first round of edits on lab04

noahstier committed:
more lab 4 edits

ykharitonova reviewed (changes requested):
Yay! Thank you for adding the lab as Markdown!
Could you please include the acknowledgement for the original lab?

ykharitonova commented on _lab/lab04.md:
```diff
@@ -0,0 +1,89 @@
+* Convert an integer to a binary representation
+* Decode a message from your instructor :-)
+
+![Blowfish](blowfish.jpg)
```
I don't think we need this image in the lab. I'd much rather add a visualization of the cypher on an example string/alphabet.

ykharitonova commented on _lab/lab04.md:
```diff
@@ -0,0 +1,89 @@
+
+Note that for this algorithm to work, the letters of the alphabet (and any other characters that should be allowed in the plaintext), as well as their order, must be fixed. For example, if you encrypt with the 26 letter English alphabet and decrypt with the 27 letter Spanish alphabet (including ñ), you won't get the correct decrypted plaintext back.
+
+Also note! Our version of the Caesar cipher has a twist: the binary key. Read on...
```
We should add another section first, since it looks like this is a good opportunity for them to get a more straightforward decryption working first (i.e., a const offset for the shift), before they work on the binary key.

ykharitonova commented on _lab/lab04.md:
```diff
@@ -0,0 +1,89 @@
+
+A Caesar cipher is one of the simplest and most widely known encryption algorithms. In this algorithm, each letter of the plaintext is replaced by a letter some fixed number of positions later in the alphabet. For example, with a shift number of 4, A would be replaced by E, B would become F, and so on. Later in this step, you will figure out how to write a decryption algorithm for the Caesar ciph ...
+
+Note that for this algorithm to work, the letters of the alphabet (and any other characters that should be allowed in the plaintext), as well as their order, must be fixed. For example, if you encrypt with the 26 letter English alphabet and decrypt with the 27 letter Spanish alphabet (including ñ), you won't get the correct decrypted plaintext back.
```
"as well as their **order**, must be fixed and the same for both encryption and decryption".

ykharitonova commented on _lab/lab04.md:
```diff
@@ -0,0 +1,89 @@
+    - create binKey
+
+    To do these you will call the helper functions you created in steps 1-2.
+Then you will have to use a for loop to go through the characters of your plaintext and for each character you will have to perform the following procedure. Pick the next character of the binKey (if the length of the binKey is smaller than the length of the plainText, then you have to cycle back around from the beginning). If the character in binKey is ‘0’ then you will have to choose the charact ...
```
We need backticks around the key to connect it to the `key` argument of the function.
Also, fwd/backward might need to be bolded.

noahstier committed:
replace blowfish image with cipher visualization

noahstier committed:
lab4 formatting

noahstier committed:
lab04 acknowledgement

noahstier committed:
split up step 4

into a simpler step about just the caesar cipher,
and a modification step that adds the binary key

noahstier commented:
Ready for review
